### PR TITLE
Material3

### PR DESCRIPTION
**Pull request by theimpulson**
## Sumary

This PR migrates Seedvault to use Material3.

## Issues

- https://github.com/seedvault-app/seedvault/issues/644

## Test

Everything passes locally

### CODE REVIEW

**Comment by theimpulson** [comment]
Switches look really bad right now. At best I can copy the colors from datura and design from AOSP's settingsLib but unless we have gradle builds or AOSP updates the material components library, it is going to look bad.

File: app/src/main/java/com/stevesoltys/seedvault/settings/SchedulingFragment.kt [hunk 1]
```diff
@@ -42,6 +45,21 @@ class SchedulingFragment : PreferenceFragmentCompat(),
...
+                dialogFragment.show(
+                    parentFragmentManager,
+                    M3ListPreference.PREFERENCE_DIALOG_FRAGMENT_TAG
+                )
+            }
+
+            else -> super.onDisplayPreferenceDialog(preference)
+        }
+    }
+
```
Comment (grote) [inline app/src/main/java/com/stevesoltys/seedvault/settings/SchedulingFragment.kt:62]: Google is really not giving us an easier way to get an MD3 list preference?

**Comment by grote** [comment]
```
> Task :app:ktlintMainSourceSetCheck FAILED
app/src/main/java/com/stevesoltys/seedvault/settings/preference/M3ListPreference.kt:47:1 Needless blank line(s) (no-consecutive-blank-lines)
```

File: app/src/main/java/com/stevesoltys/seedvault/settings/SchedulingFragment.kt (hunk 1 above)
Comment (theimpulson) [reply app/src/main/java/com/stevesoltys/seedvault/settings/SchedulingFragment.kt:62]: Nope, none of the preferences are ready for MD3 stuff even in the latest release of the library. In Aurora Store, we have a similar class for EditTextPreference too for MD3 style.

**Comment by grote** [comment]
colors seem wrong here:

Screenshot_20240624_102718

**Comment by grote** [comment]
For some reason, the exclamation mark error icon isn't red anymore: Screenshot_20240624_103537

In dark theme it is even worse, because dark greyish.

**Comment by theimpulson** [comment]
The only issue I can see right now is MD3 switches having wrong color in dark theme.

**Review by grote — APPROVED** [review]
Awesome, thanks!

**Pull request merged**